File: exercism/python/archive/hamming/hamming.py
""" calculate the number of mutations in the DNA strand (hamming distance) """
# -*- coding: utf-8 -*-

from __future__ import unicode_literals

def distance(one='', two=''):
    """ calculate the difference between the two inputs """
    return len(''.join([a for a, b in zip(one, two) if a != b]))

File: exercism/go/archive/hamming/hamming.go
// Package hamming is the packing to calculate the 'hamming distance' in a DNA strand
package hamming

import "fmt"

// testVersion is our confirmation that we're running the right test series
const testVersion = 4

// Distance is the function that'll return the number of errors in our DNA strand!
func Distance(a, b string) (int, error) {
	failcount := 0
	if len(a) != len(b) {
		return -1, fmt.Errorf("strings are not the same length\n")
	}

	for count := 0; count < len(a); count++ {
		if a[count] != b[count] {
			failcount++
		}
	}

	return failcount, nil
}
